Example/Example/UI/Weather/WeatherViewController.swift

```diff
@@ -8,94 +8,109 @@
     
     deinit {
         print(#function)
     }
             
-    @IBAction func dismiss(_ sender: Any) {
+    @IBAction private func dismiss(_ sender: Any) {
         self.dismiss(animated: true, completion: nil)
     }
     
-    @IBAction func loadWeather(_ sender: Any?) {
-        self.activityIndicator.startAnimating()
-        weatherModel.fetchWeather(at: "tokyo", date: Date()) { result in
-            DispatchQueue.main.async {
-                self.activityIndicator.stopAnimating()
-                self.handleWeather(result: result)
-            }
-        }
-        disasterModel.fetchDisaster { (disaster) in
-            self.disasterLabel.text = disaster
-        }
+    @IBAction func reload(_ sender: Any?) {
+        loadWeather()
     }
     
-    func handleWeather(result: Result<Response, WeatherError>) {
-        switch result {
-        case .success(let response):
-            self.weatherImageView.set(weather: response.weather)
-            self.minTempLabel.text = String(response.minTemp)
-            self.maxTempLabel.text = String(response.maxTemp)
-            
-        case .failure(let error):
-            let message: String
-            switch error {
-            case .jsonEncodeError:
-                message = "Jsonエンコードに失敗しました。"
-            case .jsonDecodeError:
-                message = "Jsonデコードに失敗しました。"
-            case .unknownError:
-                message = "エラーが発生しました。"
+    private func loadWeather() {
+        showIndicator { [weak self] in
+            self?.weatherModel.fetchWeather(at: "tokyo", date: Date()) { [weak self] result in
+                self?.mainQueueScheduler.schedule() {
+                    self?.handleWeather(result: result)
+                }
             }
             
-            let alertController = UIAlertController(title: "Error", message: message, preferredStyle: .alert)
-            alertController.addAction(UIAlertAction(title: "OK", style: .default) { _ in
-                self.dismiss(animated: true) {
-                    print("Close ViewController by \(alertController)")
+            self?.disasterModel.fetchDisaster { disaster in
+                self?.mainQueueScheduler.schedule() {
+                    self?.disasterLabel.text = disaster
                 }
-            })
-            self.present(alertController, animated: true, completion: nil)
+            }
+            
+        }
+    }
+    
+    private func showIndicator(while processings: () -> Void...) {
+        let dispatchGroup = DispatchGroup()
+        
+        switchView()
+        
+        processings.forEach { processing in
+            globalQueueScheduler.schedule(group: dispatchGroup) {
+                processing()
+            }
+        }
+        
+        dispatchGroup.notify(queue: .main) { [weak self] in
+            self?.switchView()
         }
     }
+    
+    private func switchView() {
+        activityIndicator.toggleAnimation()
+        reloadButton.toggleEnabled()
```

Comment:
[NITS] こちらでも十分意味は伝わるかと
```suggestion
        reloadButton.isEnabled.toggle()
```